audnauseum/state_machine/looper.py

```diff
@@ -141,62 +142,35 @@ class Looper:
          'dest': 'None'},  # Not a transition
     ]
 
-    def __init__(self, volume=1, pan=0.5, loop=None):
+    def __init__(self, loop=None):
         self.machine = Machine(model=self, states=LooperStates,
                                initial=LooperStates.IDLE,
                                transitions=Looper.transitions,
                                ignore_invalid_triggers=True,
                                after_state_change=self.echo_state_change)
-
-        self.set_default_channels()
-
         if loop is None:
             self.loop = Loop()
         else:
             self.loop = loop
 
         self.recorder = None
         self.player = Player()
+        self.reader = WavReader(loop=self.loop)
+        self.aggregator = Aggregator(
+            loop=self.loop, player_queue=self.player.input_queue, reader=self.reader)
 
         # Create a default (empty track) loop upon startup & load it
         default_path = './resources/json/default.json'
         self.write_loop(default_path)
         self.load_loop(default_path)
 
-    def set_default_channels(self):
-        """Detects and sets the default sounddevice channels
-
-        Checks the Host APIs of the user's OS audio settings for the
-        default input and output devices.
-
-        Sets the sounddevice default channels tuple to the capabilities
-        of the input and output devices.
-
-        On Linux, the default input/output devices are often a "virtual"
-        device using ALSA and can have 32, 64, or even 128 channels. This
-        value is clamped to 2 for what the physical device can support.
-        """
-        for api in sd.query_hostapis():
-            input_device = api.get('default_input_device', None)
-            output_device = api.get('default_output_device', None)
-            if input_device and input_device >= 0 and \
-                    output_device and output_device >= 0:
-                devices = sd.query_devices()
-                input_channels = devices[input_device]['max_input_channels']
-                if input_channels > 2:
-                    # Clamp value in case of virtual device
-                    input_channels = 2
-                output_channels = devices[output_device]['max_output_channels']
-                if output_channels > 2:
-                    # Clamp value in case of virtual device
-                    output_channels = 2
-                sd.default.channels = input_channels, output_channels
-                break
-
```

Comment:
We need this function to scan the devices defined in the OS to make sure it is capable of 2 channels.

Reply:
Added back in a subsequent commit